journalbeat/checkpoint/checkpoint.go

```diff
@@ -123,6 +128,51 @@ func NewCheckpoint(file string, maxUpdates int, interval time.Duration) (*Checkp
 	return c, nil
 }
 
+// Previously the registry file was written to the root folder. It was fixed on
+// 7.x but not on 6.x. Thus, migration is needed, so users avoid losing state info.
+func (c *Checkpoint) findRegistryFile() error {
+	migratedPath := paths.Resolve(paths.Data, c.file)
+
+	fs, err := os.Stat(c.file)
+	if os.IsNotExist(err) {
+		c.file = migratedPath
+		return nil
+	} else if err != nil {
+		return fmt.Errorf("error accessing previous registry file: %+v", err)
+	}
+
+	f, err := os.Open(c.file)
+	if err != nil {
+		return err
+	}
+
+	target, err := os.OpenFile(migratedPath, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, fs.Mode())
+	if err != nil {
+		return err
+	}
+
+	if _, err := io.Copy(target, f); err != nil {
+		return err
+	}
+
+	err = f.Close()
+	if err != nil {
+		return fmt.Errorf("error closing old registry file: %+v", err)
+	}
+	err = target.Close()
+	if err != nil {
+		return fmt.Errorf("error closing new registry file: %+v", err)
+	}
```

Comment:
the copy is not 'safe' without extra call to `target.Sync()`. Some filesystems (without journaling) also require a sync operation on the parent directory in order to sync file + directory meta data.

Reply:
Donz.